third_party/nvidia/hopper/lib/Transforms/WarpSpecialization.cpp

```diff
@@ -23,15 +27,75 @@ class NVGPUWarpSpecializationPass
       NVGPUWarpSpecializationPass>::NVGPUWarpSpecializationBase;
 
   void runOnFuncOp(triton::FuncOp funcOp) {
-    if (numWarpGroups <= 1)
+    SmallVector<scf::ForOp> loops;
+    funcOp->walk([&](scf::ForOp forOp) {
+      if (forOp->hasAttr(mlir::triton::kWarpSpecializeAttrName))
+        loops.push_back(forOp);
+    });
+    if (loops.empty())
+      return;
+    // FIXME: skip warpspec if there is else block.
```

Comment:
Could you document why these workarounds are needed? It would be nice to know what would need to be fixed to enable these.